composer/loggers/file_logger.py

```diff
@@ -184,39 +169,34 @@ def artifact_name(self) -> str:
 
         return name
 
-    def batch_start(self, state: State, logger: Logger) -> None:
-        self.is_batch_interval = (int(state.timestamp.batch) + 1) % self.log_interval == 0
-
     def epoch_start(self, state: State, logger: Logger) -> None:
-        self.is_epoch_interval = (int(state.timestamp.epoch) + 1) % self.log_interval == 0
         # Flush any log calls that occurred during INIT or FIT_START
         self._flush_file(logger)
 
-    def _will_log(self, log_level: LogLevel) -> bool:
-        if log_level == LogLevel.FIT:
-            return True  # fit is always logged
-        if log_level == LogLevel.EPOCH:
-            if self.log_level < LogLevel.EPOCH:
-                return False
-            if self.log_level > LogLevel.EPOCH:
-                return True
-            return self.is_epoch_interval
-        if log_level == LogLevel.BATCH:
-            if self.log_level < LogLevel.BATCH:
-                return False
-            if self.log_level > LogLevel.BATCH:
-                return True
-            return self.is_batch_interval
-        raise ValueError(f'Unknown log level: {log_level}')
-
-    def log_data(self, state: State, log_level: LogLevel, data: Dict[str, Any]):
-        if not self._will_log(log_level):
-            return
-        data_str = format_log_data_value(data)
-        self.write(
-            f'[{log_level.name}][batch={int(state.timestamp.batch)}]: ',
-            data_str + '\n',
-        )
+    def log_traces(self, traces: Dict[str, Any]):
+        if self.should_log_traces:
+            for trace_name, trace in traces.items():
+                trace_str = format_log_data_value(trace)
+                self.write(
+                    f'[trace]: {trace_name}:',
+                    trace_str + '\n',
+                )
+
+    def log_metrics(self, metrics: Dict[str, float], step: Optional[int] = None) -> None:
+        for metric_name, metric in metrics.items():
+            metric_str = format_log_data_value(metric)
```

Comment:
separate PR: maybe rename the ungainly `format_log_data_value` into something more understandable, e.g. `format_as_string`.

Reply:
captured in CO-971